vault-tool/src/Network/VaultTool.hs

```diff
@@ -503,169 +451,48 @@ vaultMounts VaultConnection{_VaultConnection_VaultAddress, _VaultConnection_Mana
-    vaultRequestJSON _VaultConnection_Manager "GET" (vaultUrl _VaultConnection_VaultAddress "/sys/mounts/" ++ T.unpack mountPoint ++ "/tune") headers (Nothing :: Maybe ()) [200]
-    where
-    headers = [authTokenHeader _VaultConnection_AuthToken]
+vaultMountTune :: VaultConnection Authenticated -> Text -> IO VaultMountConfigRead
+vaultMountTune conn mountPoint =
+    runVaultRequestAuthenticated conn
+        . newGetRequest
+        $ "/sys/mounts/" <> mountPoint <> "/tune"
 
 -- | <https://www.vaultproject.io/docs/http/sys-mounts.html>
-vaultMountSetTune :: VaultConnection -> Text -> VaultMountConfigWrite -> IO ()
-vaultMountSetTune VaultConnection{_VaultConnection_VaultAddress, _VaultConnection_Manager, _VaultConnection_AuthToken} mountPoint mountConfig = do
-    let reqBody = mountConfig
-    _ <- vaultRequest _VaultConnection_Manager "POST" (vaultUrl _VaultConnection_VaultAddress "/sys/mounts/" ++ T.unpack mountPoint ++ "/tune") headers (Just reqBody) [200]
-    pure ()
-    where
-    headers = [authTokenHeader _VaultConnection_AuthToken]
+vaultMountSetTune :: VaultConnection Authenticated -> Text -> VaultMountConfigWrite -> IO ()
+vaultMountSetTune conn mountPoint mountConfig =
+    runVaultRequestAuthenticated_ conn
+        . withStatusCodes [200, 204]
+        $ newPostRequest ("/sys/mounts/" <> mountPoint <> "/tune") (Just mountConfig)
 
 -- | <https://www.vaultproject.io/docs/http/sys-mounts.html>
-vaultNewMount :: VaultConnection -> Text -> VaultMountWrite -> IO ()
-vaultNewMount VaultConnection{_VaultConnection_VaultAddress, _VaultConnection_Manager, _VaultConnection_AuthToken} mountPoint vaultMount = do
-    let reqBody = vaultMount
-    _ <- vaultRequest _VaultConnection_Manager "POST" (vaultUrl _VaultConnection_VaultAddress "/sys/mounts/" ++ T.unpack mountPoint) headers (Just reqBody) [200]
-    pure ()
-    where
-    headers = [authTokenHeader _VaultConnection_AuthToken]
+vaultNewMount :: VaultConnection Authenticated -> Text -> VaultMountWrite -> IO ()
+vaultNewMount conn mountPoint vaultMount =
+    runVaultRequestAuthenticated_ conn
+        . withStatusCodes [200, 204]
+        $ newPostRequest ("/sys/mounts/" <> mountPoint) (Just vaultMount)
 
 -- | <https://www.vaultproject.io/docs/http/sys-mounts.html>
-vaultUnmount :: VaultConnection -> Text -> IO ()
-vaultUnmount VaultConnection{_VaultConnection_VaultAddress, _VaultConnection_Manager, _VaultConnection_AuthToken} mountPoint = do
-    _ <- vaultRequest _VaultConnection_Manager "DELETE" (vaultUrl _VaultConnection_VaultAddress "/sys/mounts/" ++ T.unpack mountPoint) headers (Nothing :: Maybe ()) [200]
-    pure ()
-    where
-    headers = [authTokenHeader _VaultConnection_AuthToken]
+vaultUnmount :: VaultConnection Authenticated -> Text -> IO ()
+vaultUnmount conn mountPoint =
+    runVaultRequestAuthenticated_ conn
+        . withStatusCodes [200, 204]
+        . newDeleteRequest
+        $ "/sys/mounts/" <> mountPoint
 
 data VaultSecretMetadata = VaultSecretMetadata
```

Comment:
It looks like this type has been duplicated in `KeyValue2`.

Reply:
I've removed this type.

Reply:
2c42064fa561f294710ccd8b3ca290ace3d42d3f